lib/matplotlib/text.py

```diff
@@ -1329,7 +1330,7 @@ def __init__(self,
                  xycoords='data',
                  annotation_clip=None):
 
-        self.xy = xy
+        self.xy = copy.copy(xy)
```

Comment:
```suggestion
        self.xy = tuple(xy)
```
Actually, we should coerce the type. Currently it's just handed through so that `Annotation.xy` might be a tuple, list ndarray, etc. depending on the user input.

Reply:
Will this correctly propagate the units with the unit tools that put units as an ndarray sub-class?

Reply:
If you give me a hint of were to find an example of this unit tools, I can make a test. @tacaswell

Reply:
I believe this discussion can be solved noting that unitful ndarrays are not a good option to pass as `xy` parameters.

I have checked packages like [unyt](https://pypi.org/project/unyt/) and it works assigning units to whole arrays. Nevertheless, the units of `xy[0]` and `xy[1]` need not to be equal (and seldom will be). If `xy` is an array it should be units-deprived, unless `x` and `y` have the same units, which is a very special case. Even so, casting to tuple in this case *does* keep the units attached to each of the tuple elements.

Moreover, a tuple *is* an acceptable and logical way of carrying unitful information in this context: `xy = (1*unyt.unit1, 1*unyt.unit2, )`.

Given that all the uses of `_AnnotationBase.xy` in the code just accesses `self.xy[0]` and `self.xy[1]` I propose either:
a) Cast the input to a tuple. I can't think of a valid `xy` input that would break in this case, or
b) Create a tuple of items in the same way they would be accessed later:
```
    self.xy = (xy[0], xy[1], )
```

@tacaswell, @timhoffm  I'll put this and foregoing pull requests on hold waiting for your input.

Reply:
To get this moving, I'm fine with the copy approach. I suppose tuple will work, but then again not 100% sure if all edge cases still do. Let's keep type coersion for another time.